src/librustc_typeck/diagnostics.rs

```diff
@@ -4092,6 +4092,110 @@ impl SpaceLlama for i32 {
 ```
 "##,
 
+E0527: r##"
+The number of elements in an array or slice pattern differed from the number of
+elements in the array being matched.
+
+Example of erroneous code:
+
+```compile_fail,E0527
+#![feature(slice_patterns)]
+
+fn main() {
+    let r = &[1, 2, 3, 4];
+    match r {
+        &[a, b] => { // error: pattern requires 2 elements but array
+                     //        has 4
+            println!("a={}, b={}", a, b);
+        }
+    }
+}
+```
+
+Ensure that the pattern is consistent with the size of the matched
+array. Additional elements can be matched with `..`:
+
+```ignore
+fn main() {
+    let r = &[1, 2, 3, 4];
+    match r {
+        &[a, b, ..] => { // ok!
+            println!("a={}, b={}", a, b);
+        }
+    }
+}
+```
+"##,
+
+E0528: r##"
+An array or slice pattern required more elements than were present in the
+matched array.
+
+Example of erroneous code:
+
+```compile_fail,E0528
+#![feature(slice_patterns)]
+
+fn main() {
+    let r = &[1, 2];
+    match r {
+        &[a, b, c, rest..] => { // error: pattern requires at least 3
+                                //        elements but array has 2
+            println!("a={}, b={}, c={} rest={:?}", a, b, c, rest);
+        }
+    }
+}
+```
+
+Ensure that the matched array has at least as many elements as there are
```

Comment:
This isn't very explicit. What about: "Ensure that the matched array has the same number of elements. If you don't need to bind them all, just add `..` or `variable..`"?